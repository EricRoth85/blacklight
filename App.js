'use strict';
import { View, StyleSheet } from 'react-native';
import React, { Component } from 'react';
import { Provider, connect } from 'react-redux';
import store from './store';

import { AppLoading, Asset, Font } from 'expo';
import { Feather, MaterialIcons, FontAwesome, Entypo } from '@expo/vector-icons';

import Router from './Router';

export default class App extends Component {

  state = {
    isLoadingComplete: false,
  };

  _loadResourcesAsync = async () => {
    return Promise.all([
      Asset.loadAsync([
        require('./images/wallpaper.jpg'),
        require('./images/city.jpg'),
        require('./images/blacklightIcon.png'),
        require('./images/light.jpg'),
      ]),
      Font.loadAsync({
        ...Ionicons.font,
        ...Feather.font,
        ...MaterialIcons.font,
        ...FontAwesome.font,
        ...Entypo.font,
      }),
    ]);
  };

  _handleLoadingError = error => {
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };


  render() {
    if (!this.state.isLoadingComplete) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <Provider store={store}>
          <View style={{ flex: 1 }}>
            <Router />
          </View>
        </Provider>
      );
    }
  }
}
