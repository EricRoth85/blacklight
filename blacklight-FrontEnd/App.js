import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { StackNavigator } from 'react-navigation';
import { Provider } from 'react-redux';

import FrontPage from './FrontPage';
import MemoryList from './MemoryList';
import MemoryMap from './MemoryMap';
import store from './store';



const Views = StackNavigator({
  Home: { screen: FrontPage },
  MemoryList: { screen: MemoryList },
  MemoryMap: { screen: MemoryMap}
});

export default class App extends Component {
  render() {

    return (
      <Provider store={store}>
        <View style={styles.container}>
          <Views />
        </View>
      </Provider>
    );

  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
