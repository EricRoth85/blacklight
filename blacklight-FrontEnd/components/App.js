'use strict';
import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { StackNavigator, TabNavigator, TabRouter } from 'react-navigation';
import { Provider } from 'react-redux';
import store from '../store';
import { Icon } from 'react-native-elements';


import FrontPage from './FrontPage';
import MemoryList from './MemoryList';
import Home from './Home';
import User from './User';
import MemoryMap from './MemoryMap';
import Login from './Login'
// const MyApp = TabRouter({
//   Home: {screen: Home},
//   User: {screen: User}
// })
import SingleMemoryView from './SingleMemoryView';


//memories stack nav config
const Views = StackNavigator({
  MemoryList: {
    screen: MemoryList


  },
  SingleMemoryView: {
    screen: SingleMemoryView
  }
});

//tabs configuration
const TabNav = TabNavigator({

  Home: {
    screen: Home,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => <Icon name="home" size={35} color={tintColor}/>
    }
  },
  Memories: {
    screen: Views,
    navigationOptions: {
      tabBarLabel: "Memories",
      tabBarIcon: ({ tintColor }) => <Icon name="menu" size={35} color={tintColor}/>
    }
  },
  Map: {
    screen: MemoryMap,
    navigationOptions: {
      tabBarLabel: "Map",
      tabBarIcon: ({ tintColor }) => <Icon name="map" size={35} color={tintColor}/>
    }
  },
  User: {
    screen: User,
    navigationOptions: {
      tabBarLabel: "User",
      tabBarIcon: ({ tintColor }) => <Icon name="account-circle" size={35} color={tintColor}/>
    }
  },
  Login: { screen: Login } // can I throw this here and have it render??

}, {
  tabBarPosition: 'bottom',
  // animationEnabled: true,
  // tabBarOptions: {
  //   style: {position: 'absolute', flex: 1},
  //   activeTintColor: '#e91e63',
  // },
});


export default class App extends Component {
  render() {

    return (
      <Provider store={store}>
        <View style={styles.container}>



          <TabNav />
        </View>
      </Provider>
    );

  }
}


const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
