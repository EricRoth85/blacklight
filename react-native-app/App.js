'use strict';
import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { StackNavigator, TabNavigator, TabRouter } from 'react-navigation';
import { Provider } from 'react-redux';
import store from './store';
import { Icon } from 'react-native-elements';

import MemoryList from './components/MemoryList';
import Home from './components/Home';
import User from './components/User';
import MemoryMap from './components/MemoryMap';
import AddMemory from './components/AddMemory';
import Login from './components/Login';

import SingleMemoryView from './components/SingleMemoryView';


//memories stack nav config
const Views = StackNavigator({
  MemoryList: {
    screen: MemoryList


  },
  SingleMemoryView: {
    screen: SingleMemoryView
  }
});

//tabs configuration
const TabNav = TabNavigator({

  Home: {
    screen: Home,
    navigationOptions: {
      tabBarLabel: "Home",
      tabBarIcon: ({ tintColor }) => <Icon name="home" size={35} color={tintColor} />
    }
  },
  Memories: {
    screen: Views,
    navigationOptions: {
      tabBarLabel: "Memories",
      tabBarIcon: ({ tintColor }) => <Icon name="menu" size={35} color={tintColor} />
    }
  },
  Map: {
    screen: MemoryMap,
    navigationOptions: {
      tabBarLabel: "Map",
      tabBarIcon: ({ tintColor }) => <Icon name="map" size={35} color={tintColor} />
    }
  },
  Post: {
    screen: AddMemory,
    navigationOptions: {
      tabBarLabel: "Post Memory",
      tabBarIcon: ({ tintColor }) => <Icon name="note-add" size={35} color={tintColor} />
    }
  },
  Login: { screen: Login } // can I throw this here and have it render??

}, {
    tabBarPosition: 'bottom',
    // animationEnabled: true,
    // tabBarOptions: {
    //   style: {position: 'absolute', flex: 1},
    //   activeTintColor: '#e91e63',
    // },
  });


export default class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <View style={styles.container}>
          <TabNav />
        </View>
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  }
});
