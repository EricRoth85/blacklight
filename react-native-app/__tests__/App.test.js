'use strict';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

describe('App component', () => {

  const AppComponent = renderer.create(<App />);
  
  it('renders AppLoading component initially', () => {
    expect(AppComponent.toJSON()).toMatchSnapshot();
  });
  
  it('renders Login component after asset loading completes', () => {
    AppComponent.getInstance()._handleFinishLoading();
    expect(AppComponent.toJSON()).toMatchSnapshot();
  });

})