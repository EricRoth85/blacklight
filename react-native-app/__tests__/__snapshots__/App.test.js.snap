// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders AppLoading component initially 1`] = `<ExponentAppLoading />`;

exports[`App component renders Login component after asset loading completes 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        onMoveShouldSetResponder={[Function]}
        onMoveShouldSetResponderCapture={[Function]}
        onResponderEnd={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderReject={[Function]}
        onResponderRelease={[Function]}
        onResponderStart={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "white",
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "flex": 1,
                "height": undefined,
                "justifyContent": "center",
                "padding": 20,
                "width": undefined,
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": undefined,
                    "width": undefined,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {},
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    Object {
                      "color": "white",
                      "fontFamily": "LaoSangamMN",
                      "fontSize": 34,
                      "fontWeight": "bold",
                      "marginTop": 10,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                BLACKLIGHT
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 15,
                    "marginRight": 15,
                  },
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "padding": 10,
                    },
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  pointerEvents="box-only"
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#9E9E9E",
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "padding": 19,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Object {
                        "backgroundColor": "#000000",
                      },
                      undefined,
                      Object {
                        "padding": 12,
                      },
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {},
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "color": "white",
                            "fontSize": 20,
                          },
                          undefined,
                          Object {
                            "fontSize": 17.5,
                          },
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                        ],
                      ]
                    }
                  >
                    CONTINUE AS ANONYMOUS
                  </Text>
                </View>
              </View>
            </View>
            <View
              keyboardVerticalOffset={0}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "padding": 10,
                  },
                  Object {
                    "paddingBottom": 0,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 15,
                          "fontWeight": "bold",
                          "marginBottom": 1,
                          "marginLeft": 20,
                          "marginRight": 20,
                          "marginTop": 15,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  Email
                </Text>
              </View>
              <View
                collapsable={undefined}
                style={
                  Object {
                    "borderBottomColor": "#bdc6cf",
                    "borderBottomWidth": 1,
                    "marginLeft": 20,
                    "marginRight": 20,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  placeholder="Email"
                  style={
                    Array [
                      Object {
                        "color": "#86939e",
                        "minHeight": 36,
                        "width": 750,
                      },
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {},
                    undefined,
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {},
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 15,
                          "fontWeight": "bold",
                          "marginBottom": 1,
                          "marginLeft": 20,
                          "marginRight": 20,
                          "marginTop": 15,
                        },
                        undefined,
                        undefined,
                      ],
                    ]
                  }
                >
                  PW
                </Text>
              </View>
              <View
                collapsable={undefined}
                style={
                  Object {
                    "borderBottomColor": "#bdc6cf",
                    "borderBottomWidth": 1,
                    "marginLeft": 20,
                    "marginRight": 20,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <TextInput
                  allowFontScaling={true}
                  onChangeText={[Function]}
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "color": "#86939e",
                        "minHeight": 36,
                        "width": 750,
                      },
                      Object {
                        "fontSize": 17.5,
                      },
                      undefined,
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "padding": 10,
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "#000000",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 20,
                            },
                            undefined,
                            Object {
                              "fontSize": 17.5,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      LOGIN
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "padding": 10,
                      },
                    ]
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "#000000",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 20,
                            },
                            undefined,
                            Object {
                              "fontSize": 17.5,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      SIGNUP
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#000000",
          "borderBottomColor": "#828287",
          "borderBottomWidth": 0.5,
          "height": 64,
          "left": 0,
          "paddingTop": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View>
        <View
          collapsable={undefined}
          style={
            Object {
              "left": 0,
              "marginTop": 10,
              "position": "absolute",
              "right": 0,
              "top": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            lineBreakMode="tail"
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "fontSize": 18,
                "left": 0,
                "opacity": 1,
                "right": 0,
                "textAlign": "center",
                "width": 180,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
